Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device_usb_wfssrv.cpp

```diff
@@ -0,0 +1,266 @@
+// Copyright 2016 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Core/IPC_HLE/WII_IPC_HLE_Device_usb_wfssrv.h"
+
+#include <string>
+#include <vector>
+
+#include "Common/CommonTypes.h"
+#include "Common/FileUtil.h"
+#include "Common/Logging/Log.h"
+#include "Common/NandPaths.h"
+#include "Core/HW/DVDInterface.h"
+#include "Core/HW/Memmap.h"
+
+namespace WFS
+{
+std::string NativePath(const std::string& wfs_path)
+{
+  return File::GetUserPath(D_WFSROOT_IDX) + Common::EscapePath(wfs_path);
+}
+}
+
+CWII_IPC_HLE_Device_usb_wfssrv::CWII_IPC_HLE_Device_usb_wfssrv(u32 device_id,
+                                                               const std::string& device_name)
+    : IWII_IPC_HLE_Device(device_id, device_name)
+{
+  m_device_name = "msc01";
+}
+
+IPCCommandResult CWII_IPC_HLE_Device_usb_wfssrv::Open(u32 command_address, u32 mode)
+{
+  INFO_LOG(WII_IPC_HLE, "/dev/usb_wfssrv: Open");
+
+  // TODO(wfs): Proper GetTitleId() implementation which takes into account
+  // nandloader content too. Refactor with ES.
+  u64 title_id;
+  DVDInterface::GetVolume().GetTitleID(&title_id);
+
+  Memory::Write_U32(title_id >> 32, command_address + 4);
+  m_Active = true;
+  return GetDefaultReply();
+}
+
+IPCCommandResult CWII_IPC_HLE_Device_usb_wfssrv::Close(u32 command_address, bool force)
```

Comment:
Since this doesn't do much more than just set m_Active to false, Close() doesn't have to be reimplemented as this is something the base class already does.